docs/src/pages/getting-started/faq/faq.md

```diff
@@ -336,3 +336,65 @@ If you use Material-UI in a commercial project and would like to support its con
 or in a side or hobby project and would like to become a backer, you can do so through [OpenCollective](https://opencollective.com/material-ui).
 
 All funds raised are managed transparently, and Sponsors receive recognition in the README and on the Material-UI home page.
+
+## Why does component X require a DOM node in a prop instead of a ref object?
+
+Components like the [Portal](/api/Portal/#props) or [Popper](/api/Popper/#props) require a DOM node in the `container` or `anchorEl` prop respectively.
+It seems convenient to simply pass a ref object in those props and let Material-UI access the current value.
+This works in a simple scenario:
+
+```jsx
+function App() {
+  const container = React.useRef(null);
+
+  return (
+    <div className="App">
+      <Portal container={container}>
+        <span>portaled children</span>
+      </Portal>
+      <div ref={container} />
+    </div>
+  );
+}
+```
+
+where `Portal` would only mount the children into the container when `container.current` is available.
+With a simple heuristic Portal would re-render after it mounts because refs are up-to-date before any effects run.
+
+```jsx
+function Portal({ children, container }) {
+  const [node, setNode] = React.useState(null);
+
+  React.useEffect(() => {
+    setNode(container.current);
+  }, [container]);
+
+  if (node === null) {
+    return null;
+  }
+  return ReactDOM.createPortal(children, node);
+}
+```
+
+This assumption can introduce subtle bugs the farther away the actual host component with the container ref is.
```

Comment:
What assumption?

Reply:
Will reword. The assumption always holds (refs are up-to-date before effects run) but just because a ref is up-to-date doesn't mean it points to an actual instance.